### PR TITLE
Remove gn_bootstrap.sh

### PR DESCRIPTION
**Pull request by andy31415**
This file was misleading because it actually runs `activate.sh` and it is generally a shell that just invokes another shell.

Now we run actual bootstrap for CI rather than activate.sh, which should ensure an up to date environment and avoid warnings of 'bootstrap done on a different machine'

It looks like the penalty of running a full bootstrap instead of activate.sh is about 80 seconds. While this is better than the old 5+ minutes of bootstrap, it is still somewhat slower with the benefit of likely a more "correct" build environment.

### CODE REVIEW

File: examples/chef/README.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ chef_$PLATFORM:
               scripts/checkout_submodules.py --allow-changing-global-git-config --shallow --platform $PLATFORM
         - name: Bootstrap
           timeout-minutes: 25
-          run: scripts/build/gn_bootstrap.sh
+          run: bash -c "source scripts/bootstrap.sh"
```
Comment (bzbarsky-apple) [inline examples/chef/README.md:153]: Why "source" in the subshell, if it's going to exit anyway?  That is, why not just `bash scripts/bootstrap.sh`?

File: integrations/docker/images/chip-cert-bins/Dockerfile [hunk 2]
```diff
@@ -12,187 +12,187 @@ ARG ZAP_VERSION=v2023.04.27-nightly
 # Ensure TARGETPLATFORM is set
 RUN case ${TARGETPLATFORM} in \
         "linux/amd64") \
-            echo "Building for linux/amd64" \
+        echo "Building for linux/amd64" \
```
Comment (bzbarsky-apple) [inline integrations/docker/images/chip-cert-bins/Dockerfile:15]: Why the whitespace changes?

File: integrations/docker/images/chip-cert-bins/Dockerfile [hunk 3]
```diff
@@ -201,12 +201,12 @@ RUN git clone https://github.com/project-chip/connectedhomeip.git /root/connecte
 WORKDIR /root/connectedhomeip/
 RUN git checkout ${COMMITHASH}
 RUN ./scripts/checkout_submodules.py --allow-changing-global-git-config --shallow --platform linux
-RUN scripts/build/gn_bootstrap.sh
+RUN bash -c "source scripts/bootstrap.sh"
```
Comment (bzbarsky-apple) [inline integrations/docker/images/chip-cert-bins/Dockerfile:204]: Same question here as in the YAML.

**Review by bzbarsky-apple — APPROVED** [review]

File: integrations/docker/images/chip-cert-bins/Dockerfile (hunk 2 above)
Comment (andy31415) [reply integrations/docker/images/chip-cert-bins/Dockerfile]: Some auto-formatter ran. Undone the formatting.

**Review by arkq — APPROVED** [review]

**Pull request merged**